pages/admin/treatments/components/TreatmentModal.tsx

```diff
@@ -0,0 +1,252 @@
+
+    useEffect(() => {
+        if (isOpen) {
+            setErrors({});
+        }
+        if (isOpen && mode === 'edit' && selectedTreatment) {
+            setFormData({
+                name: selectedTreatment.name,
+                description: selectedTreatment.description,
+                price: selectedTreatment.price,
+                duration: selectedTreatment.duration,
+            });
+            return;
+        }
+        if (mode === 'create') {
+            setFormData({
+                name: '',
+                description: '',
+                price: 0,
+                duration: 0,
+            });
+        }
+    }, [isOpen, mode, selectedTreatment]);
+
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const { name, value, type } = e.target;
+        // clean errors
+        setErrors((prevErrors) => {
+            const newErrors = { ...prevErrors };
+            delete newErrors[name];
+            return newErrors;
+        });
+        setFormData({
+            ...formData,
+            [name]: type === 'number' ? Number(value) : value,
+        });
+    };
+
+    const handleSubmit = () => {
+        if (mode === 'create' || mode === 'edit') {
+            // Field validation
+            const fieldErrors = validateTreatmentFields(formData);
+            if (Object.keys(fieldErrors).length > 0) {
+                setErrors(fieldErrors);
+                return;
+            }
+            onSubmit(formData);
+        } else if (mode === 'delete') {
+            onSubmit();
+        }
+    };
+
+    const getTitle = () => {
+        switch (mode) {
+            case 'create':
+                return 'Nuevo tratamiento';
+            case 'edit':
+                return 'Editar tratamiento';
+            case 'delete':
+                return 'Eliminar tratamiento';
+            default:
+                return 'Tratamiento';
+        }
+    };
+
+    return (
+        <Dialog.Root
+            open={isOpen}
+            onOpenChange={({ open }) => !open && onClose()}
+            placement='center'
+        >
+            <Portal>
+                <Dialog.Backdrop />
+                <DialogPositioner>
+                    <Dialog.Content p={5} backgroundColor='white'>
+                        <Dialog.Header
+                            mb={4}
+                            display='flex'
+                            justifyContent='space-between'
+                            alignContent='center'
+                        >
+                            <Dialog.Title>{getTitle()}</Dialog.Title>
+                            <Dialog.CloseTrigger asChild>
+                                <CloseButton color='black' size='sm' />
+                            </Dialog.CloseTrigger>
+                        </Dialog.Header>
+
+                        <Dialog.Body>
+                            {mode === 'create' || mode === 'edit' ? (
+                                <form>
```

Comment:
💭 I think the form can be in a separated component, that way you can handle their inner logic there